backend/spec/requests/users_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe("Users") do
+  describe "POST /users/" do
+    it "returns user object with token if params are valid" do
+      post("/users/", params: {
+        user: {
+          username: "bhoss_killah_mapagmahal",
+          password: "p@ssw0rd",
+        },
+      })
+      expect(response).to(have_http_status(:created))
+      user = json_body[:data][:user]
+      decoded_token = JsonWebToken.decode(json_body[:data][:token])
+
+      expect(user[:id]).to(eql(decoded_token["user_id"]))
+      expect(json_body[:errors]).to(be_nil)
+    end
+
+    it "returns error if username params already exists" do
+      post("/users/", params: {
+        user: {
```

Comment:
Yeah as Normz mentioned you can improve this by using let. Madami actually way to structure tests, some points you can review are

let vs let! (basically usage of bang operator in general)
subject block